scalyr_agent/log_processing.py

```diff
@@ -597,7 +597,7 @@ def readline(self, current_time=None):
 
             except Exception, e:
                 # something went wrong. Return the full line and log a message
-                log.warn("Error parsing line as json for %s.  Logging full line: %s\n%s" % (self.__path, str(e), result.line.decode( "utf-8", 'replace' )),
+                log.warn("Error parsing line as json for log '%s' - %s" % (self.__path, e),
```

Comment:
1. The comment is correct.  We return the full line (rather than the JSON parsed line) and then log a message saying that an error occurred.

2. We don't eliminate customer log content.  The full raw log still gets logged to the main log for that file (and will therefore go through the appropriate redaction and other rules).  It's just no longer *also* being output to the agent.log.  The debug message contains the name of the log file that had the problem, so it's possible to look at that log at the appropriate point in time and pinpoint which raw line caused the issue.